rfcs/0095-enable-docheck-by-default.md

```diff
@@ -0,0 +1,69 @@
+---
+feature: enable-docheck-by-default
+start-date: 2021-06-24
+author: gytis-ivaskevicius
+co-authors:
+shepherd-team: @Ericson2314 @nh2 @edolstra
+shepherd-leader: @Ericson2314
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+Enable `doCheck` by default when using `stdenv.mkDerivation` function.
+
+# Motivation
+[motivation]: #motivation
+
+I believe that an additional quality gate would be beneficial to the derivations build process.
+
+Resolution of this RFC is expected to remove [these comments](https://github.com/NixOS/nixpkgs/blob/8c563eaf7049d82fbe95b0847ac5ae6e5554e2fa/pkgs/stdenv/generic/make-derivation.nix#L61-L67)
+either by enabling `checkPhase` by default or rejecting this RFC.
+
+# Detailed design
+[design]: #detailed-design
+
+The basic idea is quite simple
+- New `doCheck`/`doInstallCheck` semantic should be implemented.
+- By default `doCheck` option should be enabled as long as `stdenv.hostPlatform == stdenv.buildPlatform`.
+- Non-reproducible test prevention should be implemented.
+- All failing packages should be fixed or updated with `doCheck = false;`
+
+**New `doCheck`/`doInstallCheck` semantics:**
+In addition to booleans, `doCheck`/`doInstallCheck` should also accept strings.
+- String value should be considered as `false`
+- It should be used as a place for comment on why the check is disabled. For
+  example: "Requires X11 server" or "Requires network access".
```

Comment:
```suggestion
**New semantics:**
- `doCheck`/`doInstallCheck` should default to `null` and work exactly the same as `false` 
- New options should be introduced: `meta.{checksFlaky,checksLargeDependencies,checksTakeTooLong,checksDisableReason}`
```

The first point is so we could evaluate checks that are not set. Derivation paths are not expected to change.
And the second point is so we could evaluate the reason why checks were disabled. Maybe defining additional attrset would be nice? like `checks.{flaky,largeDependencies,takeTooLong,disableReason}`? Any thoughts?

Reply:
A comment on the source code and a Boolean value shoud suffice.

Reply:
During RFC call, we thought that it would be a very simple change which would allow us to actually evaluate the reason why tests were disabled and that is definitely handy. I am still in favor of it

Reply:
That being said, it is preferrable a (short?) string, without true/false semantics attached to it. 
After all, it can be useful to use the string to convey a useful information even in the case the tests are mandatory.

Something like 

```nix
doCheck = false;
checkReason = "I don't wanna fetch X.Org just to verify a blinking box!";
```

Or even better, a new attrset:

```nix
check.enable = true;
check.reason = "This package is critical for bootstrap";
```